include/deal.II/multigrid/mg_base.h

```diff
@@ -189,7 +189,22 @@ class MGTransferBase : public Subscriptor
   virtual void
   prolongate(const unsigned int to_level,
              VectorType &       dst,
-             const VectorType & src) const = 0;
+             const VectorType & src) const;
```

Comment:
One thing I'm wondering is whether we should leave this function pure virtual and let the other function `prolongate_and_add` simply use `prolongate` followed by a vector addition here in the base class. This would allow us to arrange things the other way around, i.e., no changes in all other transfer classes except the `MGTransferMatrixFree` (and possibly the one for global coarsening). I think this is less code overall and slightly simpler to understand.

Reply:
The reason why I went this way is that in the approach you describing we would need to allocated a temporal vector.

Reply:
The problem is that `prolongate_and_add` is the less common function, and `vmult_add` internally also allocates temporary vectors for some of our matrix classes, e.g. the Trilinos matrices https://github.com/dealii/dealii/blob/e4171ed1729d09c842a6ee77c08bd7e669c5fd1c/source/lac/trilinos_sparse_matrix.cc#L2060-L2070

Therefore I think we should go with the other option, which is also slightly simpler to understand.